### PR TITLE
Allow subscription type in laterpay_redeem_voucher shortcode

### PR DESCRIPTION
**Pull request by dhaval-parekh**
Reference Issue 1348.

### CODE REVIEW

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 1]
```diff
@@ -415,29 +415,50 @@ public function ajax_get_premium_shortcode_link( LaterPay_Core_Event $event ) {
...
+
+        list( $atts ) = $event->get_arguments() + array( array() );
 
         $data = shortcode_atts( array(
-            'id' => null,
+            'id'   => null,
+            'type' => 'timepass',
         ), $atts );
 
+        $allowed_types = [ 'time-pass', 'subscription' ];
```
Comment (thrijith) [inline laterpay/application/Controller/Frontend/Shortcode.php]:
```suggestion
        $allowed_types = [ 'timepass', 'subscription' ];
```

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 2]
```diff
@@ -415,29 +415,50 @@ public function ajax_get_premium_shortcode_link( LaterPay_Core_Event $event ) {
...
+            if ( 'subscription' === $data['type'] ) {
+                $pass_data = LaterPay_Helper_Subscription::get_subscription_by_id( $data['id'], true );
+            } else {
+                $pass_data = LaterPay_Helper_TimePass::get_time_pass_by_id( $data['id'], true );
+            }
+
+            if ( ! $pass_data ) {
+
+                if ( 'subscription' === $data['type'] ) {
+                    $error_message = __( 'Wrong subscription id.', 'laterpay' );
```
Comment (thrijith) [inline laterpay/application/Controller/Frontend/Shortcode.php]:
```suggestion
                    $error_message = __( 'Wrong Subscription ID.', 'laterpay' );
```

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 3]
```diff
@@ -415,29 +415,50 @@ public function ajax_get_premium_shortcode_link( LaterPay_Core_Event $event ) {
...
+                } else {
+                    $error_message = __( 'Wrong time pass id.', 'laterpay' );
```
Comment (thrijith) [inline laterpay/application/Controller/Frontend/Shortcode.php]:
```suggestion
                    $error_message = __( 'Wrong Time Pass ID.', 'laterpay' );
```

File: laterpay/views/frontend/partials/post/gift/gift-pass.php [hunk 4]
```diff
@@ -22,9 +22,12 @@
 if ( 0 !== absint( $gift_pass['access_to'] ) ) {
     $access_dest = $category->name;
 }
+
+$pass_id = ( ! empty( $gift_pass['pass_id'] ) ) ? $gift_pass['pass_id'] : '';
+$pass_id = ( empty( $pass_id ) && ! empty( $gift_pass['id'] ) ) ? $gift_pass['id'] : '';
 ?>
 
-<div class="lp_js_giftCard lp_gift-card lp_gift-card-<?php echo esc_attr( $gift_pass['pass_id'] ); ?>">
+<div class="lp_js_giftCard lp_gift-card lp_gift-card-<?php echo esc_attr( $pass_id ); ?>">
```
Comment (thrijith) [inline laterpay/views/frontend/partials/post/gift/gift-pass.php:34]: Should the card be displayed if `$pass_id` is empty in the end?

**Review by thrijith — DISMISSED** [review]

File: laterpay/application/Controller/Frontend/Shortcode.php (hunk 1 above)
Comment (dhaval-parekh) [reply laterpay/application/Controller/Frontend/Shortcode.php]: Done.

File: laterpay/application/Controller/Frontend/Shortcode.php (hunk 3 above)
Comment (dhaval-parekh) [reply laterpay/application/Controller/Frontend/Shortcode.php]: Done

File: laterpay/views/frontend/partials/post/gift/gift-pass.php (hunk 4 above)
Comment (dhaval-parekh) [reply laterpay/views/frontend/partials/post/gift/gift-pass.php:34]: Added condition. if `$pass_id` is empty don't render template.

File: laterpay/application/Controller/Frontend/Shortcode.php (hunk 2 above)
Comment (dhaval-parekh) [reply laterpay/application/Controller/Frontend/Shortcode.php]: Done

**Pull request closed without merging**